src/axom/quest/MeshViewUtil.hpp

```diff
@@ -0,0 +1,778 @@
+    m_topology = &m_dom->fetch_existing(topoPath);
+
+    const std::string coordsetPath =
+      "coordsets/" + m_dom->fetch_existing(topoPath + "/coordset").as_string();
+    m_coordset = &m_dom->fetch_existing(coordsetPath);
+
+    m_cdom = m_dom;
+    m_ctopology = m_topology;
+    m_ccoordset = m_coordset;
+
+    if(!isValid())
+    {
+      SLIC_ERROR(
+        "Invalid domain in MeshViewUtil.  Domain must be blueprint-valid,"
+        " structured, with exlicit coordinates and dimension equal to "
+        " the template DIM parameter.");
+    }
+
+    computeCoordsDataLayout();
+  }
+
+  /*!
+    @brief Constructor
+    @param [in] bpDomain const Blueprint single domain.
+    @param [in] topologyName Name op topology in the domain.
+  */
+  MeshViewUtil(const conduit::Node& bpDomain, const std::string& topologyName)
+    : m_cdom(&bpDomain)
+    , m_topologyName(topologyName)
+  {
+    const std::string topoPath = "topologies/" + m_topologyName;
+    SLIC_ASSERT(m_cdom->has_path(topoPath));
+    m_ctopology = &m_cdom->fetch_existing(topoPath);
+
+    const std::string coordsetPath =
+      "coordsets/" + m_cdom->fetch_existing(topoPath + "/coordset").as_string();
+    m_ccoordset = &m_cdom->fetch_existing(coordsetPath);
+
+    if(!isValid())
+    {
+      SLIC_ERROR(
+        "Invalid domain in MeshViewUtil.  Domain must be blueprint-valid,"
+        " structured, with exlicit coordinates. and dimension equal to "
+        " the template DIM parameter.");
+    }
+
+    computeCoordsDataLayout();
+  }
+
+  /*!
+    @brief Whether the domain is a valid one for this utility.
+
+    In addition to checking for blueprint validity (unless \a
+    checkBlueprint is false), this method checks against its own
+    requirements.
+  */
+  bool isValid(bool checkBlueprint = false) const
+  {
+    bool rval = true;
+    if(checkBlueprint)
+    {
+      conduit::Node info;
+  #ifdef AXOM_USE_MPI
+      rval = rval &&
+        conduit::blueprint::mpi::verify("mesh", *m_cdom, info, MPI_COMM_WORLD);
+  #else
+      rval = rval && conduit::blueprint::verify("mesh", *m_cdom, info);
+  #endif
+    }
+    rval =
+      rval && (m_ctopology->fetch_existing("type").as_string() == "structured");
+    rval =
+      rval && (m_ccoordset->fetch_existing("type").as_string() == "explicit");
+    rval =
+      rval && (conduit::blueprint::mesh::coordset::dims(*m_ccoordset) == DIM);
+    return true;
+  }
+  //@}
+
+  //@{
+  //!@name Blueprint data organization
+  /*!
+    @brief Get the named topology.
+
+    @return the Conduit node at "topologies/<topologyName>".
+  */
+  const conduit::Node& getTopology() { return *m_topology; }
+  const conduit::Node& getTopology() const { return *m_ctopology; }
+
+  /*!
+    @brief Get the coordinates conduit::Node for the named topology.
+  */
+  conduit::Node& getCoordSet() { return *m_coordset; }
+  const conduit::Node& getCoordSet() const { return *m_ccoordset; }
```

Comment:
Note: If `m_coordset` were a path to a conduit node, these could return  `parent_node[ m_coordset_path ]`

Reply:
Agreed, but see previous comment.